modules/desktop/profiles/applications.nix

```diff
@@ -4,21 +4,112 @@
 {
   config,
   lib,
+  pkgs,
   ...
 }: let
   cfg = config.ghaf.profiles.applications;
-in {
-  options.ghaf.profiles.applications = {
-    enable = lib.mkEnableOption "Some sample applications";
-    #TODO Create options to allow enabling individual apps
-  };
+in
+  with lib; {
+    options.ghaf.profiles.applications = {
+      enable = mkEnableOption "Some sample applications";
+      #TODO Create options to allow enabling individual apps
+      #weston.ini.nix mods needed
+      ivShMemServer = {
+        enable = mkOption {
+          type = types.bool;
+          default = false;
+          description = "Enable memory sharing between virtual machines";
+        };
+        memSize = mkOption {
+          type = lib.types.int;
+          default = 16;
+          description = mdDoc ''
+            Defines shared memory size
+          '';
+        };
+        vmCount = mkOption {
```

Comment:
I have some concerns with `vmCount` used as a duplicate limit for already declared number of application VMs.
If/when we must(?) set this as [a build-time parameter to memsocket](https://github.com/tiiuae/ghaf/pull/473/files#diff-4cd0792a1806df6e4de8a4de75cf451ec4536470b07fa5b4a3f10ad041ed989eR31), could we instead [reuse the length of `enabled-app-vms` list](https://github.com/tiiuae/ghaf/blob/2db4ceeff045a8bc4c858bb1ff33481f70ed7c26/modules/reference/appvms/default.nix#L22) to follow DRY-principle?

As a practical question - what happens if the the `vmCount` is declared smaller than the amount of system enabled application VMs?

Reply:
> As a practical question - what happens if the the `vmCount` is declared smaller than the amount of system enabled application VMs?
1. Interrupts from excessive VMs will not interrupt the peer as it allocates only the `vmCount` pairs of interrupts.
2. The `memsocket` application will fail as there is insufficient shared memory for excessive VMs.

Reply:
> [reuse the length of enabled-app-vms list](https://github.com/tiiuae/ghaf/blob/2db4ceeff045a8bc4c858bb1ff33481f70ed7c26/modules/reference/appvms/default.nix#L22) to follow DRY-principle?

Done.